### PR TITLE
feat: add new migration guides [netlify-build]

### PR DESCRIPTION
**Pull request by alanjcharles**
### Proposed changes

added new mobile migration guides

### Merge timing
 as soon as we have team sign-off

### CODE REVIEW

File: src/connections/sources/catalog/libraries/mobile/apple/migration.md [hunk 1]
```diff
@@ -1,27 +1,22 @@
 ---
 title: Analytics for Swift Migration Guide
 strat: swift
-tags:
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/apple/migration.md:4]: @alanjcharles Need to leave these in for searchability

File: src/connections/sources/catalog/libraries/mobile/apple/migration.md [hunk 2]
```diff
@@ -45,24 +40,73 @@ If you're using a different mobile library such as Analytics-iOS, follow these s
...
     
     _analytics = [[SEGAnalytics alloc] initWithConfiguration: config];
 ```
 {% endcodeexampletab %}
 {% endcodeexample %}
 
-## Convert Middlewares to Plugins
+
+> success ""
+> NOTE: Analytics-Swift supports running multiple instances of the analytics object, so it does not assume a singleton. However, if you’re migrating from Analytics-iOS and all your track calls are routed to the Analytics.shared() singleton, you can very easily forward these calls to your new Analytics-swift object.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/apple/migration.md:56]:
```suggestion
> Analytics-Swift supports running multiple instances of the analytics object, so it does not assume a singleton. However, if you’re migrating from Analytics-iOS and all your track calls are routed to the `Analytics.shared()` singleton, you can these calls to your new Analytics-swift object.
```

File: src/connections/sources/catalog/libraries/mobile/apple/migration.md [hunk 3]
```diff
@@ -45,24 +40,73 @@ If you're using a different mobile library such as Analytics-iOS, follow these s
...
+
+With this simple extension, all of your existing Segment calls should work with Analytics-Swift.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/apple/migration.md:58]:
```suggestion
With this extension, your existing Segment calls should work with Analytics-Swift. 
```

File: src/connections/sources/catalog/libraries/mobile/apple/migration.md [hunk 4]
```diff
@@ -45,24 +40,73 @@ If you're using a different mobile library such as Analytics-iOS, follow these s
...
+```swift
+@extension Analytics {
+    (SegAnalytics)shared() {
+        return analytics; // or whatever variable name you're using
+    } 
+}
+```
+## 2. Upgrade your Destinations 
+
+If your app uses Segment to route data to Destinations via Segment-cloud (i.e. Cloud-mode destinations), you can skip this step. Analytics-Swift treats Device-mode Destinations as [plugins](/docs/connections/sources/catalog/libraries/mobile/swift/plugin-architecture), and simplifies the process in integrating them into your app. Analytics-Swift supports these [Device-Mode Destinations](/docs/connections/sources/catalog/libraries/mobile/swift/destination-plugins) with more to come.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/apple/migration.md:69]:
```suggestion
If your app uses Segment to route data to Destinations through Segment-cloud (for example, Cloud-mode destinations), you can skip this step. Analytics-Swift treats Device-mode Destinations as [plugins](/docs/connections/sources/catalog/libraries/mobile/swift/plugin-architecture), and simplifies the process of integrating them into your app. Analytics-Swift supports these [Device-Mode Destinations](/docs/connections/sources/catalog/libraries/mobile/swift/destination-plugins). 
```

File: src/connections/sources/catalog/libraries/mobile/apple/migration.md [hunk 5]
```diff
@@ -45,24 +40,73 @@ If you're using a different mobile library such as Analytics-iOS, follow these s
...
+        ),
+```
+### 2.b) Add Plugin to your Analytics instnace
+
+```
+import Segment
+import Segment<Destination> // <-- Add this line
+```
+
+Just under your Analytics-Swift library setup, call `analytics.add(plugin: ...)` to add an instance of the plugin to the Analytics timeline.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/apple/migration.md:87]:
```suggestion
Under your Analytics-Swift library setup, call `analytics.add(plugin: ...)` to add an instance of the plugin to the Analytics timeline.
```

File: src/connections/sources/catalog/libraries/mobile/apple/migration.md [hunk 6]
```diff
@@ -187,154 +339,20 @@ Segment added these options:
...
-```
-
-**After example**
-<br> 
-
-```swift    
-    analytics.alias(newId: "new id")
-```
+### Conclusion
+Hopefully this simplifies your migration to Analytics-Swift. Once you’re up and running, you can take advantage of Analytics-Swift’s additional features, such as Destination Filters, Functions & Typewriter support.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/apple/migration.md:358]:
```suggestion
Once you’re up and running, you can take advantage of Analytics-Swift’s additional features, such as [Destination Filters](/docs/connections/sources/catalog/libraries/mobile/apple/swift-destination-plugins), [Functions](/docs/connections/functions/), and [Typewriter](/docs/connections/sources/catalog/libraries/mobile/apple/swift-typewriter) support. 
```

File: src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md [hunk 7]
```diff
@@ -7,99 +7,120 @@ tags:
...
+    import com.segment.analytics.kotlin.android.AndroidAnalyticsKt; // For calling from Android
+    import com.segment.analytics.kotlin.core.compat.JavaAnalytics; // For calling from Java
+    import com.segment.analytics.kotlin.core.platform.Plugin; // Replaces Middleware
+  ```
+  {% endcodeexampletab %}
+  {% endcodeexample %}
+
+## 2. Upgrade your Destinations
+
+If your app uses Segment to route data to Destinations via Segment-cloud (i.e. Cloud-mode destinations), you can skip this step. Analytics-Kotlin treats Device-mode Destinations as plugins, and simplifies the process in integrating them into your app. Analytics-Kotlin supports these Device-Mode Destinations with more to come.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md:87]:
```suggestion
If your app uses Segment to route data to Destinations through Segment-cloud (for example, Cloud-mode destinations), you can skip this step. Analytics-Kotlin treats Device-mode Destinations as plugins, and simplifies the process in integrating them into your app. Analytics-Kotlin supports these Device-Mode Destinations with more to come.
```

File: src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md [hunk 8]
```diff
@@ -7,99 +7,120 @@ tags:
...
+  ```
+  {% endcodeexampletab %}
+  {% endcodeexample %}
+
+Your events will now begin to flow to the added destination in Device-Mode.
+
+## 3. Upgrade Middleware to Plugins
 
-Middlewares are a powerful mechanism that can augment the events collected by the SDK. A middleware is a function that the Segment SDK invokes and can be used to monitor, modify, augment or reject events.
+Middlewares are a powerful mechanism that can augment events collected by the Analytics iOS (Classic) SDK. A middleware is a simple function that is invoked by the Segment SDK and can be used to monitor, modify, augment or reject events. Analytics Swift replaces the concept of middlewares with Enrichment Plugins to give you even more control over your event data. Refer to the Plugin Architecture Overview for more information.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md:114]: @alanjcharles Should this paragraph reference Kotlin?

File: src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md [hunk 9]
```diff
@@ -327,505 +348,56 @@ After example:
...
+Java
+Map map = new HashMap<String, Object>();
+JsonElement json = JSONKt.toJsonElement(map);
 
-Options are no longer supported and should be converted into plugins.
\ No newline at end of file
+### 4.c) Options
+Options are no longer supported and should be converted into plugins.
+## Conclusion
+Hopefully this simplifies your migration to Analytics-Kotlin. Once you’re up and running, you can take advantage of Analytics-Kotlin’s additional features, such as Destination Filters, Functions & Typewriter support.
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md:403]:
```suggestion
Once you’re up and running, you can take advantage of Analytics-Kotlin’s additional features, like [Destination Filters](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/kotlin-android-destination-filters/), [Functions](https://segment.com/docs/connections/functions/), and [Typewriter](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/kotlin-android-typewriter/) support.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

File: src/connections/sources/catalog/libraries/mobile/apple/migration.md [hunk 10]
```diff
@@ -1,27 +1,22 @@
 ---
 title: Analytics for Swift Migration Guide
 strat: swift
```
Comment (markzegarelli) [inline src/connections/sources/catalog/libraries/mobile/apple/migration.md:3]:
```suggestion
strat: swift
tags:
  - apple
  - swift
  - ios
```

File: src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md (hunk 8 above)
Comment (alanjcharles) [reply src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md:114]: @markzegarelli oh, yeah definitely

**Pull request merged**